File: app/api/src/main/java/com/somba/api/core/enums/Category.java
package com.somba.api.core.enums;

import com.somba.api.core.exceptions.InvalidCategoryException;
import com.somba.api.core.exceptions.NullCategoryException;

public enum Category {
    ELECTRONICS,
    CLOTHES,
    SHOES,
    HOME,
    BEAUTY,
    SPORTS,
    OTHERS;



    public static Category fromValue(String value) {
      if (value == null) {
          throw new NullCategoryException();
      }
        for (Category category : Category.values()) {
            if (category.name().equalsIgnoreCase(value)) {
                return category;
            }
        }
        throw new InvalidCategoryException(value);
    }
}

File: app/ui/lib/domain/enums/category.dart

enum Category {
  all,
  electronics,
  clothing,
  shoes,
  home,
  beauty,
  sports,
}

extension CategoryExtension on Category {
  String get name {
    switch (this) {
      case Category.all:
        return 'All';
      case Category.electronics:
        return 'Electronics';
      case Category.clothing:
        return 'Clothes';
      case Category.home:
        return 'Home';
      case Category.beauty:
        return 'Beauty';
      case Category.sports:
        return 'Sports';
      case Category.shoes:
        return 'Shoes';
      default:
        return '';
    }
  }
}
